### PR TITLE
[ui] select by form/expression harmonization and feature parity

### PR DESCRIPTION
**Pull request by nirvn**
@nyalldawson , this one's for you. I've taken some time to try and harmonize the select by form & select by expression dialogs by doing the following:
- adding a zoom to feature function to the select by expression dialog
- updating the wording of the selection action in the select by expression dialog
- adding icons to the select by form dialog's select push button
- re-ordering the selection action in the select by form dialog

Select by form:
f

Select by expression:
e

### CODE REVIEW

File: src/gui/qgsexpressionselectiondialog.cpp [hunk 1]
```diff
@@ -103,6 +118,63 @@ void QgsExpressionSelectionDialog::on_mActionRemoveFromSelection_triggered()
...
+    bbox.combineExtentWith( r );
+    featureCount++;
+  }
+  features.close();
+
+  QSettings settings;
+  int timeout = settings.value( QStringLiteral( "/qgis/messageTimeout" ), 5 ).toInt();
+  if ( featureCount > 0 )
+  {
+    mMapCanvas->zoomToFeatureExtent( bbox );
```
Comment (nyalldawson) [inline src/gui/qgsexpressionselectiondialog.cpp]: mMapCanvas may be null here. I'd check for this and return at the start of the function.

File: python/gui/qgsexpressionselectiondialog.sip [hunk 2]
```diff
@@ -35,11 +35,25 @@ class QgsExpressionSelectionDialog : QDialog
...
+     * @note added in QGIS 3.0
+     */
+    void setMapCanvas( QgsMapCanvas* canvas );
+
   public slots:
     void on_mActionSelect_triggered();
     void on_mActionAddToSelection_triggered();
     void on_mActionRemoveFromSelection_triggered();
     void on_mActionSelectIntersect_triggered();
+    void on_mButtonZoomToSelection_clicked();
```
Comment (nyalldawson) [inline python/gui/qgsexpressionselectiondialog.sip]: I think this should be private - I don't like how all these slots are public and become part of stable api for no good reason.

**Comment by nyalldawson** [comment]
Looks great, nice tweaks!

File: src/gui/qgsexpressionselectiondialog.cpp (hunk 1 above)
Comment (nirvn) [inline src/gui/qgsexpressionselectiondialog.cpp]: Good point; I'll do a null check, and hide the "zoom to features" button if no map canvas is set.

File: python/gui/qgsexpressionselectiondialog.sip (hunk 2 above)
Comment (nirvn) [inline python/gui/qgsexpressionselectiondialog.sip]: +1, I'll make all of the on_* private.

**Pull request merged**